src/rabbit_shovel_parameters.erl

```diff
@@ -194,12 +194,16 @@ validate_params_user(#amqp_params_direct{}, none) ->
     ok;
 validate_params_user(#amqp_params_direct{virtual_host = VHost},
                      User = #user{username = Username}) ->
-    case rabbit_vhost:exists(VHost) andalso
-        (catch rabbit_access_control:check_vhost_access(
-                 User, VHost, undefined, #{})) of
+    VHostAccess = case catch rabbit_access_control:check_vhost_access(User, VHost, undefined, #{}) of
+                      ok -> ok;
+                      NotOK ->
+                          rabbit_log:debug("rabbit_access_control:check_vhost_access result: ~p", [NotOK]),
+                          NotOK
+                  end,
+    case rabbit_vhost:exists(VHost) andalso VHostAccess of
```

Comment:
`Vhost` access can be either `ok` or of type `rabbit_types:channel_exit()`. Can we use `andalso` with either of those?

Reply:
We can! I was surprised as well - https://erlang.org/doc/reference_manual/expressions.html#short-circuit-expressions

As re-written the code is exactly equal to what was there before, just clearer in my opinion.